### PR TITLE
Rotating platform

### PR DESCRIPTION
**Pull request by LMiljak**
The platform now rotates depending on how synchronous the Android devices are moving.

### CODE REVIEW

File: ContextDesktop/src/main/java/com/github/migi_1/Context/model/entity/behaviour/PlatformRotateBehaviour.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+			Vector3f res = new Vector3f(
+					(float) (Math.cos(time * 2) * amplitude), 
+					0.0f,
+					(float) (Math.cos(time) * amplitude)
+				);
+			
+			super.setRotateVector(res);
+		}
+	}
+
```
Comment (dVoorhout) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/entity/behaviour/PlatformRotateBehaviour.java:68]: Please leave some comments on this method, it's a little unclear what it exactly does and why.

File: ContextDesktop/src/main/java/com/github/migi_1/Context/model/entity/behaviour/PlatformRotateBehaviour.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		
+		Collection<Vector3f> vectors = new ArrayList<>(carrierBehaviours.size());
+		
+		for (MoveBehaviour behaviour : carrierBehaviours) {
+			vectors.add(behaviour.getMoveVector());
+		}
+		
+		return aggregator.aggregate(vectors);
+	}
+
```
Comment (dVoorhout) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/entity/behaviour/PlatformRotateBehaviour.java:86]: same here

File: ContextDesktop/src/main/java/com/github/migi_1/Context/utility/DistanceVectorAggregator.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+		ArrayList<Vector3f> newVectors = new ArrayList<>(vectors.size());
+		for (Vector3f vector : vectors) {
+			Vector3f subtracted = average.subtract(vector);
+			newVectors.add(subtracted.mult(subtracted));
+		}
+		
+		Vector3f mean = averager.aggregate(newVectors);
+		return new Vector3f((float) Math.sqrt(mean.x), (float) Math.sqrt(mean.y), (float) Math.sqrt(mean.z));
+	}
+
```
Comment (dVoorhout) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/utility/DistanceVectorAggregator.java:36]: and here

Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/utility/DistanceVectorAggregator.java:36]: See the Javadoc of the class, it's explained there.

**Comment by LMiljak** [comment]
Added comments where necesarry. Thanks Damian

File: ContextDesktop/src/main/java/com/github/migi_1/Context/model/entity/Platform.java [hunk 4]
```diff
@@ -44,7 +43,13 @@ public Platform(Vector3f startLocation, MainEnvironment environment) {
         super();
         
         this.carrierAssigner = new CarrierAssigner(this, Main.getInstance().getServer(), environment);
+
+        ArrayList<AccelerometerMoveBehaviour> b = new ArrayList<>(4);
+        for (PlatformPosition p : PlatformPosition.values()) {
+        	b.add(new AccelerometerMoveBehaviour(ip -> ip.equals(carrierAssigner.getAddress(p))));
+        }
```
Comment (marcelkuipers) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/entity/Platform.java:52]: b and p are not very descriptive variable names

**Comment by marcelkuipers** [comment]
Besides my comment this PR looks fine!

**Comment by LMiljak** [comment]
:+1:  Renamed them. I was being hasty.

**Comment by NilsHullegien** [comment]
Looks good to me!

**Pull request merged**